### PR TITLE
bump cni library version to v0.8.0

### PR DESCRIPTION
**Pull request by aojea**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
/kind cleanup

> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test

/kind feature
/kind flake

**What this PR does / why we need it**:

last CNI library release is 0.7.1 from Jun 11, 2019.
Since then, there was introduced new feature and bugfixes.

Currently, this library is only being used by dockershim,
the other CRI plugins are vendoring it directly

However, this will help also to mitigate some of the issues with the
CI jobs that are still using dockershim.

Specially interesting the new cache feature and some bugfixes to improve
resilience to errors.

Signed-off-by: Antonio Ojea <antonio.ojea.garcia@gmail.com>

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
@aojea let's please ask them to tag 0.7.2 for us to use.

/hold

**Comment by aojea** [comment]
poke to cni maintainer to see if they can tag a new release
/cc @squeed

**Comment by bboreham** [comment]
> let's please ask them to tag 0.7.2 for us to use.

Opening an issue in the CNI repo would be a good way to do that.

I believe it would be 0.8.0 due to a (minor) interface change.

**Comment by aojea** [comment]
@dims updated to latest cni release
thanks @bboreham

**Comment by dims** [comment]
/retest

@liggitt this looks ready!

File: vendor/github.com/containernetworking/cni/pkg/invoke/raw_exec.go
Comment (liggitt) [inline vendor/github.com/containernetworking/cni/pkg/invoke/raw_exec.go:57]: is there a path in kube that hits this?

Comment (aojea) [reply vendor/github.com/containernetworking/cni/pkg/invoke/raw_exec.go:57]: @saschagrunert you've added this https://github.com/containernetworking/cni/pull/763, 
can you help me to answer this quesiton :smile: ?

Comment (saschagrunert) [reply vendor/github.com/containernetworking/cni/pkg/invoke/raw_exec.go:57]: There was a setup race where the init container downloads the CNI Plugin binary (like Cilium) and the runtime immediately picks up the (not finished copying) executable on disk. This caused issues in multiple distributions on cluster bootstrap, so we added this feature of retrying for sake of laziness.

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**
